src_features/getPublicKey/cmd_getPublicKey.c

```diff
@@ -35,25 +34,19 @@ void handleGetPublicKey(uint8_t p1,
     }
 
     tmpCtx.publicKeyContext.getChaincode = (p2 == P2_CHAINCODE);
-    io_seproxyhal_io_heartbeat();
-    os_perso_derive_node_bip32(
-        CX_CURVE_256K1,
-        bip32.path,
-        bip32.length,
-        privateKeyData,
-        (tmpCtx.publicKeyContext.getChaincode ? tmpCtx.publicKeyContext.chainCode : NULL));
-    cx_ecfp_init_private_key(CX_CURVE_256K1, privateKeyData, 32, &privateKey);
-    io_seproxyhal_io_heartbeat();
-    cx_ecfp_generate_pair(CX_CURVE_256K1, &tmpCtx.publicKeyContext.publicKey, &privateKey, 1);
-    explicit_bzero(&privateKey, sizeof(privateKey));
-    explicit_bzero(privateKeyData, sizeof(privateKeyData));
-    io_seproxyhal_io_heartbeat();
-    if (!getEthAddressStringFromKey(&tmpCtx.publicKeyContext.publicKey,
-                                    tmpCtx.publicKeyContext.address,
-                                    &global_sha3,
-                                    chainConfig->chainId)) {
-        THROW(CX_INVALID_PARAMETER);
+    if (bip32_derive_get_pubkey_256(
+            CX_CURVE_256K1,
+            bip32.path,
+            bip32.length,
+            tmpCtx.publicKeyContext.publicKey.W,
+            (tmpCtx.publicKeyContext.getChaincode ? tmpCtx.publicKeyContext.chainCode : NULL),
+            CX_SHA512) != CX_OK) {
+        THROW(0x6F00);
```

Comment:
do we have a defined apdu for this ?

Reply:
What do you mean, instead of the error code 0x6F00? I could rather use a `CX_ASSERT`?